lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -147,12 +147,20 @@
 )
 
 VM_SIZE_FALLBACK_PATTERNS = [
-    # exclude Standard_DS1_v2, because one core is too slow,
-    # and doesn't work in some distro
-    re.compile(r"Standard_DS((?!1)[\d])_v2"),
-    re.compile(r"Standard_DS([\d]{2})_v2"),
+    # First priority: Standard_D series with single digit
+    # (excluding D1) because one core is too slow, and doesn't work in some distro
+    # e.g., Standard_DS2_v2, Standard_D2_v5, Standard_D4s_v3
+    re.compile(r"^Standard_D[A-Z]*((?!1)[\d])[a-z]*_v\d+$"),
+    # Second Priority, remaining D series VM Sizes with two digit core count
```

Comment:
Please explain the difference between `D[\d]+` and `D[A-Z][\d]+` in the code comment.

Reply:
Is the below comment sufficient? 

'D[A-Z]*[\d]+' is used as instead of 'D[\d]+' to also pick up older VM sizes like 'Standard_DS5_v2'

Reply:
or do you think we should just have D[\d]+, and not prioritize the older vm size? This would simplify the pattern